submissions/okkkko/js-post-oop/index.js

```diff
@@ -0,0 +1,74 @@
+class Inhabitant {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  toString() {
+    const properties = ["species", "name", "gender", "saying", "friends"].map(
+      (prop) => `${prop}: ${this[prop]}`
+    );
+    return properties.join("; ");
+  }
+}
+class Human extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    super("human", name, gender, saying, friends);
+    this.legs = 2;
+    this.hands = 2;
+  }
+  toString() {
+    return `${super.toString()}; ` + `hands: ${this.hands}; legs:${this.legs};`;
+  }
+}
+class Man extends Human {
+  constructor(name, saying, friends) {
+    super(name, "male", saying, friends);
+  }
+}
+class Woman extends Human {
+  constructor(name, saying, friends) {
+    super(name, "female", saying, friends);
+  }
+}
+class Dog extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    super("dog", name, gender, saying, friends);
+    this.legs = 4;
+    this.hands = 0;
+  }
+  toString() {
+    return `${super.toString()}; ` + `hands: ${this.hands}; legs:${this.legs};`;
+  }
+}
+class Cat extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    super("cat", name, gender, saying, friends);
+    this.legs = 4;
```

Comment:
All inhabitants have legs. Just like they all have names and gender etc.
Why handling legs in a different way?

Reply:
There are could be inhabitants without legs, and it would be more difficult to maintain code

Reply:
This doesn't answer the question.
There could be inhabitants without gender or names. But current tech spec provides for named, gendered and legged inhabitants.

Reply:
I give up) I don`t know, I thought that it could be too specific to add legs to Inhabitant class, but maybe in this context it is appropriate